opm/grid/LookUpData.hh

```diff
@@ -277,6 +308,42 @@ Opm::LookUpData<Grid,GridView>::operator()(const EntityType& elem,
     return fieldProp[fieldPropIdx];
 }
 
+template<typename Grid, typename GridView>
+std::vector<double> Opm::LookUpData<Grid,GridView>::assignFieldPropsDoubleOnLeaf(const FieldPropsManager& fieldPropsManager,
+                                                                                 const std::string& propString,
+                                                                                 const unsigned int& numElements,
+                                                                                 const double& defaultValue) const
+{
+    std::vector<double> fieldPropOnLeaf;
+    fieldPropOnLeaf.resize(numElements, defaultValue);
+    const auto& fieldProp = fieldPropsManager.get_double(propString);
+    for (unsigned int elemIdx = 0; elemIdx < numElements; ++elemIdx) {
+        const auto& fieldPropIdx = this->getFieldPropIdx<Grid>(elemIdx);
+        fieldPropOnLeaf[elemIdx] = fieldProp[fieldPropIdx];
+    }
+    return fieldPropOnLeaf;
+}
+
+template<typename Grid, typename GridView>
+template<typename IntType>
+std::vector<IntType> Opm::LookUpData<Grid,GridView>::assignFieldPropsIntOnLeaf(const FieldPropsManager& fieldPropsManager,
+                                                                               const std::string& propString,
+                                                                               const unsigned int& numElements,
+                                                                               const int& defaultValue,
+                                                                               const bool& needsTranslation,
+                                                                               std::function<void(IntType)> valueCheck) const
+{
+    std::vector<IntType> fieldPropOnLeaf;
+    fieldPropOnLeaf.resize(numElements, defaultValue);
```

Comment:
See other PR, Maybe it seems more consistent to also do the translation on default value.

Reply:
See my comment below. I don't get what the translation on the default value means, but it might make sense to get the right default value via the name of the field property.